src/db.rs

```diff
@@ -0,0 +1,67 @@
+use chrono::serde::ts_seconds_option;
+use serde::{Deserialize, Serialize};
+use sqlx::{
+    types::chrono::{DateTime, Utc},
+    FromRow, PgPool,
+};
+
+///Represents a go link, a mapping from source -> sink with some metadata
+#[derive(Debug, FromRow, Serialize, Deserialize)]
+pub struct Redirect {
+    pub source: String,
+    pub sink: String,
+    ///number of times this link has been used
+    pub usages: i32,
+
+    ///Last time this link was used. Serialized as seconds since epoch
+    #[serde(with = "ts_seconds_option")]
+    pub last_used: Option<DateTime<Utc>>,
+
+    ///When this link was created. Serialized as seconds since epoch
+    #[serde(with = "ts_seconds_option")]
+    pub created: Option<DateTime<Utc>>,
+}
+
+/// Gets the redirect URL (sink) for a given source
+pub async fn get_sink(source: &str, conn: &PgPool) -> Result<Option<String>, sqlx::Error> {
+    sqlx::query!("SELECT sink FROM redirects_new WHERE source = $1", source)
+        .fetch_optional(conn)
+        .await
+        .map(|s| s.map(|s| s.sink))
+}
+
+/// Gets the n most recently used redirects.
+pub async fn get_recent(conn: &PgPool, n: i64) -> Result<Vec<Redirect>, sqlx::Error> {
```

Comment:
how will this be used? will we be paginating recently used?

Reply:
If like an API user wants to paginate them then maybe?

Reply:
Should we then have a eg. Get 101-200th most recent used? Otherwise later pages will need to fetch more

Reply:
This is a good point. I clearly do not know how pagination works.